qt/widgets/mplcpp/src/Axes.cpp

```diff
@@ -131,6 +131,29 @@ void Axes::setTitle(const char *label) {
   callMethodNoCheck<void, const char *>(pyobj(), "set_title", label);
 }
 
+/**
+ * Format the tick labels on an axis or axes
+ * @param char* axis :: [ 'x' | 'y' | 'both' ]
+ * @param char* style :: [ 'sci' (or 'scientific') | 'plain' ] plain turns off
+ * scientific notation
+ * @param bool useOffset :: True, the offset will be
+ * calculated as needed, False no offset will be used
+ */
+void Axes::tickLabelFormat(const char *axis, const char *style,
+                           const bool useOffset) {
+  try {
+    GlobalInterpreterLock lock;
```

Comment:
It's needed. It's to do with threading I think. Basically when you declare the lock you automatically acquire the lock and then when the code goes out of scope it automatically releases the lock. Without it the code does not work correctly.

Reply:
Just to add some context: The Global Interpreter Lock (GIL) in Python is a global mutex to ensure global data structures in the Python libraries are consistent across multiple threads. You can read more at https://realpython.com/python-gil/.

The use of an object in C++ to manage this is useful because we can guarantee the lock is released however the function exits, via return or exception. This pattern in C++ is called [RAII](https://en.cppreference.com/w/cpp/language/raii) and you typically see it used when acquiring resources such as memory, files sockets etc. The big example in the standard library is `std::unique_ptr` - it acquires memory on construction and guarantees it is freed.